Custom block code and msg

shashank11p opened the pull request:
Adding a FilterResult in the Filter API, so that filters can provide custom blocking status code and message along with the FilterResult. 
Using this FilterResult to block and set the response status and message accordingly

varkey98 commented on filter-api/src/main/java/org/hypertrace/agent/filter/MultiFilter.java:
```diff
@@ -48,16 +49,16 @@ public boolean evaluateRequestHeaders(Span span, Map<String, String> headers) {
       }
     }
-    return shouldBlock;
+    return new FilterResult(false, 403, "Hypertrace Blocked Request");
```
the default message and status code would be 200 and OK right? since we 're setting shouldBlock to false

varkey98 replied:
yes its a nit only, I get that we would be overriding this in our filter implementation and incase of pass through it wont be set at all, but just for consistency

shashank11p replied:
Updated to empty msg and `0` as status code since they should not be used

ryanericson reviewed (commented):
I suppose this is pending discussion for GRPC

shashank11p commented:
> I suppose this is pending discussion for GRPC

Yes, for now it's setting the Status description as the msg provided. I'll just revert that in grpc, and we will update grpc when we finalise how to set it for grpc.

shashank11p commented:
> > I suppose this is pending discussion for GRPC
> 
> Yes, for now it's setting the Status description as the msg provided. I'll just revert that in grpc, and we will update grpc when we finalise how to set it for grpc.

Reverted these. Right now, we do not add custom status and msgs for grpc. 
We have added them for all other frameworks that we support
